x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/pytorch/process/PyTorchResultProcessor.java

```diff
@@ -107,17 +107,19 @@ public void process(PyTorchProcess process) {
 
                 if (result.inferenceResult() != null) {
                     processInferenceResult(result);
-                }
-                ThreadSettings threadSettings = result.threadSettings();
-                if (threadSettings != null) {
-                    threadSettingsConsumer.accept(threadSettings);
+                } else if (result.threadSettings() != null) {
```

Comment:
Each request is of a single type (inference, thread settings etc) and the response is always of a single type so changing this to `else if` should not make any difference as every result should have only 1 non null field